elements/ethernet/etherrewrite.cc

```diff
@@ -71,12 +63,39 @@ Packet *
 EtherRewrite::pull(int)
 {
     if (Packet *p = input(0).pull())
-	return smaction(p);
+       return smaction(p);
     else
-	return 0;
+       return 0;
+}
+
+#if HAVE_BATCH
+PacketBatch *
+EtherRewrite::simple_action_batch(PacketBatch * batch)
```

Comment:
Either simple_action(_batch) or the combination of pull(_batch), push_(batch) should be implemented but not the 3. Having pull+push is slightly faster than simple_action, so I would suggest removing simple action.